curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -361,19 +363,27 @@ int FileCacheManager::Read(uint64_t inodeId, uint64_t offset, uint64_t length,
             if (ret < 0) {
                 retry++;
                 responses.clear();
-                if (ret != -2 || retry == maxRetry) {
+                if (ret != -2) {
                     LOG(ERROR) << "read from s3 failed. ret:" << ret;
                     return ret;
                 } else {
-                    // ret -2 refs s3obj not exist
-                    // clear inodecache && get again
-                    LOG(INFO) << "inode cache maybe steal, try to get latest";
-                    ::curve::common::UniqueLock lgGuard =
-                        inodeWrapper->GetUniqueLock();
-                    auto r = inodeWrapper->RefreshS3ChunkInfo();
-                    if (r != CURVEFS_ERROR::OK) {
-                        LOG(WARNING) << "refresh inode fail, ret:" << ret;
-                        return -1;
+                    if (retry < maxRetry) {
+                        // ret -2 refs s3obj not exist
+                        // clear inodecache && get again
+                        LOG(INFO) << "inode cache maybe steal, try to get "
+                                     "latest, retry:"
+                                  << retry;
+                        ::curve::common::UniqueLock lgGuard =
+                            inodeWrapper->GetUniqueLock();
+                        auto r = inodeWrapper->RefreshS3ChunkInfo();
+                        if (r != CURVEFS_ERROR::OK) {
+                            LOG(WARNING) << "refresh inode fail, ret:" << ret;
+                            return -1;
+                        }
+                        bthread_usleep(retryIntervalMs * retry * 1000);
+                    } else {
+                        LOG(WARNING) << "retry is reach max retry:" << maxRetry;
+                        bthread_usleep(retryIntervalMs * maxRetry * 1000);
```

Comment:
if so, the `maxRetry` doesn't means the total number of retries, it means the max sleep intervals between retries.

Reply:
I have the same doubt.

Reply:
fixed